plugin/src/main/kotlin/org/openrewrite/gradle/AndroidProjectSpec.kt

```diff
@@ -0,0 +1,292 @@
+        propertiesFiles[name] = text
+    }
+
+    private val yamlFiles: MutableMap<String, String> = mutableMapOf()
+    fun yamlFile(name: String, @Language("yaml") text: String) {
+        yamlFiles[name] = text
+    }
+
+    private val groovyClasses: MutableList<String> = mutableListOf()
+    fun groovyClass(@Language("groovy") source: String) {
+        groovyClasses.add(source.trimIndent())
+    }
+
+
+    @Suppress("RegExpSimplifiable")
+    fun build(dir: Path): AndroidSourceSetSpec {
+        Files.createDirectories(dir)
+        for (javaSource in javaSources) {
+            val packageDecl = if (javaSource.startsWith("package")) {
+                "package\\s+([a-zA-Z0-9.]+);".toRegex(RegexOption.MULTILINE)
+                    .find(javaSource)!!
+                    .groupValues[1]
+            } else {
+                ""
+            }.replace(".", "/")
+            val clazz = ".*(class|interface|enum)\\s+([a-zA-Z0-9-_]+)".toRegex(RegexOption.MULTILINE)
+                .find(javaSource)!!.groupValues[2]
+            val path = if (packageDecl.isEmpty()) {
+                "$name/java/$clazz.java"
+            } else {
+                "$name/java/$packageDecl/$clazz.java"
+            }
+            dir.resolve(path).apply {
+                Files.createDirectories(parent)
+                Files.write(this, javaSource.toByteArray(sourceCharset))
+            }
+        }
+        for (kotlinSource in kotlinSources) {
+            val packageDecl = if (kotlinSource.startsWith("package")) {
+                "package\\s+([a-zA-Z0-9.]+)".toRegex(RegexOption.MULTILINE)
+                    .find(kotlinSource)!!
+                    .groupValues[1]
+            } else {
+                ""
+            }.replace(".", "/")
+            val clazz = ".*(class|interface|enum)\\s+([a-zA-Z0-9-_]+)".toRegex(RegexOption.MULTILINE)
+                .find(kotlinSource)!!.groupValues[2]
+            val path = if (packageDecl.isEmpty()) {
+                "$name/kotlin/$clazz.kt"
+            } else {
+                "$name/kotlin/$packageDecl/$clazz.kt"
+            }
+            dir.resolve(path).apply {
+                Files.createDirectories(parent)
+                Files.write(this, kotlinSource.toByteArray(sourceCharset))
+            }
+        }
+        for (groovySource in groovyClasses) {
+            val packageDecl = if (groovySource.startsWith("package")) {
+                "package\\s+([a-zA-Z0-9.]+);?".toRegex(RegexOption.MULTILINE)
+                    .find(groovySource)!!
+                    .groupValues[1]
+            } else {
+                ""
+            }.replace(".", "/")
+            val clazz = ".*(class|interface|enum)\\s+([a-zA-Z0-9-_]+)".toRegex(RegexOption.MULTILINE)
+                .find(groovySource)!!.groupValues[2]
+            val path = if (packageDecl.isEmpty()) {
+                "$name/groovy/$clazz.groovy"
+            } else {
+                "$name/groovy/$packageDecl/$clazz.groovy"
```

Comment:
You will probably want to replace the dots with slashes in the package name.

Reply:
This class was not needed and I've deleted it.